terraform/developer-guide.md

```diff
@@ -0,0 +1,39 @@
+## How to create and setup GCP projects using terraform
+
+### Prerequisites
+
+- Make sure [terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/gcp-get-started) is installed
+- Make sure you have access to `who-terraform-admin` project.
+- Create a new key for the `terraform@who-terraform-admin.iam.gserviceaccount.com` service account in the `who-terraform-admin` project, download the JSON credentials, and configure your environment:
+
+  ```sh
+  export GOOGLE_APPLICATION_CREDENTIALS=<path to json credentials>
+  ```
+
+- Have a GCP billing account:
+
+  ```sh
+  export TF_VAR_billing_account=<GCP billing account id>
+  ```
+
+### Create or update a GCP project
+
+- Configure environment:
+
+  ```sh
+  export TF_VAR_project_name=<project name>
+  export TF_VAR_region=<valid GCP region, e.g. europe-west6>
+  ```
+
+- Change directory into `terraform/` and do
+  ```sh
+  terraform init
+  ```
+- Deploy
+  ```sh
+  terraform apply
```

Comment:
It's probably better practice to do the following (adding `out` to `.gitignore`, of course):

```
terraform plan -out out
terraform apply out
```

so the user is exactly sure about which changes will happen before they actually get applied

Reply:
@epicfaace - Terraform by default shows you the changes plan before it's applied. I've used this many times and it's worked fine for me. The only thing that I think would be dangerous is `-auto-approve`:

```
terraform apply -auto-approve          Skip interactive approval of plan before applying.
```